modules/gdscript/gdscript_parser.h

```diff
@@ -151,24 +151,34 @@ class GDScriptParser {
 		String to_string() const;
 		PropertyInfo to_property_info(const String &p_name) const;
 
-		_FORCE_INLINE_ void set_container_element_type(const DataType &p_type) {
-			container_element_type = memnew(DataType(p_type));
+		_FORCE_INLINE_ int container_element_type_size() const {
+			return container_element_types.size();
 		}
 
-		_FORCE_INLINE_ DataType get_container_element_type() const {
-			ERR_FAIL_NULL_V(container_element_type, DataType());
-			return *container_element_type;
+		_FORCE_INLINE_ void set_container_element_type(int p_index, const DataType &p_type) {
+			while (container_element_type_size() <= p_index) {
+				container_element_types.push_back(nullptr);
+			}
+			container_element_types.set(p_index, memnew(DataType(p_type)));
 		}
 
-		_FORCE_INLINE_ bool has_container_element_type() const {
-			return container_element_type != nullptr;
+		_FORCE_INLINE_ DataType get_container_element_type(int p_index) const {
+			ERR_FAIL_COND_V(container_element_type_size() <= p_index, DataType());
```

Comment:
I have to say this condition tripped me at first because I usually see this done the other direction (if index is bigger or equal than size).

That said, might be better to use `ERR_FAIL_INDEX_V` to make it clear:
```suggestion
			ERR_FAIL_INDEX_V(p_index, container_element_type_size(), DataType());
```